src/main/kotlin/com/blincke/commune_api/repositories/UserRepository.kt

```diff
@@ -15,37 +15,71 @@ interface UserRepository : JpaRepository<User, String> {
                 "and u not in " +
                 "(select f.friendshipId.recipient from Friendship f " +
                 "where f.friendshipId.requester = (?1))" +
-                "and u != (?1)"
+                "and u != (?1) " +
+                "and lower(u.name) like %?2%"
     )
-    fun selectSuggestedFriends(user: User, pageable: Pageable): List<User>
+    fun selectSuggestedFriends(
+        user: User,
+        nameContainsLower: String,
+        pageable: Pageable,
+    ): List<User>
 
+    /**
+     * select f1.recipient
+     * from friendship f1
+     * inner join friendship f2
+     * on f1.requester = f2.recipient
+     * and f1.recipient = f2.requester
+     * and f1.requester = 'userId';
+     */
     @Query(
```

Comment:
Finds all occurences of friendships where (requester, recipient) and (recipient, requester) exist in the database and requester = userId.